corona-warn-app/cwa-app-android
# Contact Journal Risk Extension - Risk details contact overview logic & peristance (EXPOSUREAPP-4577)

**DRAFT PR.** 
Relates to: EXPOSUREAPP-4577 
As App User I would like to understand in which situations in the past I had possibly an exposure risk, so that I can adjust my behavior in the future accordingly. This personal exposure risk is added to the contact journal. The user can see on which date a low & high risk appeared.

**TODO:**
- adjust / add special testcases (e.g. migration,  clearing aggregated risk per date results, etc.)
- some code adjustments & conflicts
- device testing without fake data

**How to test (testdata & testerbuilld):**
- Enter the testmenu
- Enter Enf-v2-Calculations
- Set Exposure Windows to 'increased Risk Default'
- Calculate Risk Level by button to get a relevant risk state
- Enter the Contact Diary / Journal and watch results
- Repeat with other fake exposure windows or reset risk level by button

**How to test (realdata)**
- Take to devices 
- Chose a working CDN / Server Environment (Debug Options)
- Bring one device in "infected state" (positiv QR Code)
- Wait until the other device reports a relevant risk
- Check the contact journal for relevant entries after a period of time (instant, minutes, days)

**Screenshots:**

![contact-screen-2](https://user-images.githubusercontent.com/46747780/105037284-0ad3a400-5a5e-11eb-947e-f2148ea581be.png)
![contact-screen-1](https://user-images.githubusercontent.com/46747780/105037287-0b6c3a80-5a5e-11eb-8a1f-1bff046af9a9.png)
